components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,193 @@
+import pg from "pg";
+
 export default {
   type: "app",
   app: "postgresql",
-  propDefinitions: {},
+  propDefinitions: {
+    table: {
+      type: "string",
+      label: "Table",
+      description: "Database table",
+      async options() {
+        return this.getTables();
+      },
+    },
+    column: {
+      type: "string",
+      label: "Column",
+      description: "The name of a column in the table to use for deduplication. Defaults to the table's primary key",
+      async options({ table }) {
+        return this.getColumns(table);
+      },
+    },
+    query: {
+      type: "string",
+      label: "SQL Query",
+      description: "Your custom SQL query",
+    },
+    value: {
+      type: "string",
+      label: "Lookup Value",
+      description: "Value to search for",
+      async options({
+        table, column,
+      }) {
+        return this.getColumnValues(table, column);
+      },
```

Comment:
I'm not sure about getting all values from one table in this way because the table might have a huge number of records, so pagination would be a must here I guess. The other way I see it is by doing it with just a simple where string (e.g. [MySql delete-row](https://github.com/PipedreamHQ/pipedream/blob/ac90bf536ce0e911ac0bc7b07d6ba2e23b09e18e/components/mysql/actions/delete-row/delete-row.mjs#L27))

Reply:
Updated to use pagination.